roles/kubernetes/control_plane/init/primary/templates/kubeadm-config.j2

```diff
@@ -7,14 +7,21 @@
 #
 # kubeadm Configuration : https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta2/
 #
-{% if kubernetes_version is version('1.26.0', 'lt', version_type='semver') %}
-apiVersion: kubeadm.k8s.io/v1beta2
-{% else %}
+{% if kubernetes_version is version('1.31.0', 'lt', version_type='semver') %}
 apiVersion: kubeadm.k8s.io/v1beta3
+{% elif kubernetes_version is version('1.33.0', 'lt', version_type='semver') %}
+apiVersion: kubeadm.k8s.io/v1beta4
+{% else %}
+apiVersion: kubeadm.k8s.io/v1
 {% endif %}
 kind: ClusterConfiguration
 certificatesDir: /etc/kubernetes/pki
 kubernetesVersion: v{{ kubernetes_version }}
+clusterName: "{{ kubernetes_cluster_name }}"
+controlPlaneEndpoint: "{{ kubernetes_vip_fqdn }}:6443"
```

Comment:
Is the port static? I have seen where this can change, but not sure if that's a configurable value here.